Language List & Paradigm Comparison Overhaul

tessmr opened the pull request:
Merged ChoiceData into Guess as requested. 
Removed ChoiceData
Updated LanguageArray to have all language variables private static final
Updated GameFrame to work with the new way paradigms are referenced (paradigmID for comparisons and paradigmName for displays)
Updated GuessHandler to work with the new way paradigms are referenced and added a paradigm comparison for languages that have similar paradigms. 

When we are done testing we need to change the hardcoded correctAnswer in GameFrame to one that is created by the SelectLanguage class.

tessmr committed:
Update GuessHandler.java

Changed the return color for matchYear to be MY_YELLOW if the year is incorrect to reflect the decision we made earlier today. Also updated the comment above the line of code to reflect that change.

tessmr committed:
Added three files to come up with a correct guess for the game. 

LanguageArray.java contains a list of objects for each programming language and its data. An array contains all the language objects and there is a method to retrieve that array.  

SelectLanguage.java randomly selects a programming language from the array provided by the LanguageArray class. 

ChoiceData.java stores the randomly selected object and contains getters to easily retrieve the information for the selected language.

tessmr committed:
Delete src/model/ChoiceData.java

Deleted ChoiceData because it is duplicative of Guess

tessmr committed:
Update Guess.java

Merged features (private variables, getters, and updated paradigm to an int) from ChoiceData into Guess

tessmr committed:
Update GuessHandler.java

Updated file to handle paradigm as an int instead of a string

tessmr committed:
Update LanguageArray.java

Changing each language variable to be private static final from just private.

tessmr committed:
Update GameFrame.java

Updated to use the getters in Guess rather than the direct variable and also to work with paradigm as an int.

tessmr committed:
Update Guess.java

Updated to include a new getter getParadigmName() which can use the paradigmID to get the name for display.

tessmr committed:
Update GameFrame.java

Now uses getParadigmName() to find the correct name of the paradigm based off of its ID

tessmr committed:
Update GuessHandler.java

Updates paradigm to show when the user is in the same 'family' as the correct language's paradigm

tessmr committed:
Update GameFrame.java

Fixed a change left out in the initial update to refer to the number version of paradigm as ParadigmID

tessmr committed:
Update LanguageArray.java

Removed some old references to ChoiceData. Replaced them with Guess

tessmr committed:
Update SelectLanguage.java

Removed some old references to ChoiceData. Replaced them with Guess.

RoboRobert commented on src/model/Guess.java:
```diff
@@ -11,24 +11,24 @@
-    public String level;
-    public int year;
+    private String name;
+    private String typing;
```
Currently the program doesn't compile because you removed the `private String level;` if you add that back, everything works fine.

RoboRobert commented on src/model/Guess.java:
```diff
@@ -38,6 +38,45 @@ public Guess(String nameIn, String typingIn, String paradigmIn, String levelIn,
+        return paradigmID;
+    }
+    
+    public String getParadigmName()
+    {
+        switch(paradigmID)
+        {
+            case 1 -> paradigmName = "PROCEDURAL";
+            case 2 -> paradigmName = "STRUCTURED";
+            case 3 -> paradigmName = "OBJECT ORIENTED";
+            case 4 -> paradigmName = "FUNCTIONAL";
+            case 5 -> paradigmName = "LOGIC";
+        }
+        return paradigmName;
+    }
```
Honestly, I would flip this function around to convert from a String to an ID. IDs are less readable and should probably just be used internally for easy comparisons.

RoboRobert commented on src/model/Guess.java:
```diff
@@ -11,24 +11,24 @@
      */
-    public Guess(String nameIn, String typingIn, String paradigmIn, String levelIn, int yearIn) {
+    public Guess(String nameIn, String typingIn, int paradigmIDIn, String levelIn, int yearIn) {
         name = nameIn;
         typing = typingIn;
-        paradigm = paradigmIn;
+        paradigmID = paradigmIDIn;
         level = levelIn;
         year = yearIn;
```
I would change this Guess constructor to read in `paradigmName`, then just make a function to convert from the name to an ID, and call it in the constructor to set the ID.

RoboRobert reviewed (changes requested):
Nice work! A few small things to consider.

tessmr committed:
Update Guess.java

Added back the private String level

tessmr committed:
Update Guess.java

Swapped the ParadigmName and ParadigmIDs so our program can primarily reference the paradigms by name.

tessmr committed:
Update GuessHandler.java

Updated GuessHandler to function with paradigms as strings again

tessmr committed:
Update LanguageArray.java

Reworked the language objects to refer to paradigms by their strings

tessmr committed:
Update GameFrame.java

Updated GameFrame to reference the paradigm by name instead of ID.

tessmr commented:
Everything has been updated! Doublechecked that it compiled and made the changes you requested!
